tests/preprocessor_test.py

```diff
@@ -27,120 +98,106 @@ def preprocess_and_compare(source, reference, tmpdir, supportfiles=[], optional_
-            assert(outfile.read() == expected_file.read())
+    def test_function_and_macro_calls_2(self, tmpdir):
+        preprocess_and_compare_functionally('function_like_macro_2.cpp',
+                                            'function_like_macro_2.cpp.preprocessed.cc')
 
     def test_basic_function_with_defaults_refactored(self, tmpdir):
-        preprocess_and_compare('file_include.cpp', 'preprocessed_file_include.cpp',
-                               tmpdir, ['SomeFile.h', 'SomeOtherFile.h'])
+        preprocess_and_compare_functionally('file_include.cpp', 'preprocessed_file_include.cpp')
 
     def test_ifdef_handling(self, tmpdir):
-        preprocess_and_compare('ifdef.cpp', 'ifdef.cpp.preprocessed.cc', tmpdir)
+        preprocess_and_compare_functionally('ifdef.cpp', 'ifdef.cpp.preprocessed.cc')
+
+    def test_for_loop_not_breaking_macros(self, tmpdir):
+        preprocess_and_compare_functionally("for_loop.cpp", "for_loop.cpp.preprocessed.cc")
 
     def test_system_file_include(self, tmpdir):
-        # copy some files to the tmpdir, then run search for them
         system_dir = tmpdir.mkdir('system_include_path')
+
+        args = FakeArgs()
+        args.sys_include = [system_dir.realpath()]
+        # copy some files to the tmpdir, then run search for them
+
         shutil.copy(SOURCE_FILE_DIRECTORY + 'SomeFile.h', f"{system_dir.realpath()}/SomeFile.h")
         shutil.copy(SOURCE_FILE_DIRECTORY + 'SomeOtherFile.h',
                     f"{system_dir.realpath()}/SomeOtherFile.h")
 
-        preprocess_and_compare('systemish_include.cpp', 'systemish_include.cpp.preprocessed.cc',
-                               tmpdir, optional_args=['-S', system_dir.realpath()])
+        preprocess_and_compare_functionally('systemish_include.cpp',
+                                            'systemish_include.cpp.preprocessed.cc',
+                                            args)
 
-    def test_comments(self, tmpdir):
-        preprocess_and_compare('comments.cpp', 'comments.cpp.preprocessed.cc', tmpdir)
+    def test_include_path_search(self, tmpdir):
+        # copy some files to the tmpdir, then run search for them
+        test_dir = tmpdir.mkdir('include_path')
+        shutil.copy(SOURCE_FILE_DIRECTORY + 'SomeFile.h', test_dir.realpath())
+        symtable.SYSTEM_INCLUDE_PATHS.append(str(test_dir.realpath()))
 
-    def test_nested_macro_expansion(self, tmpdir):
-        preprocess_and_compare('multiple_macros.cpp', 'multiple_macros.cpp.preprocessed.cc', tmpdir)
+        found = ast.PreprocessorIncludeNode.search_system_includes('SomeFile.h')
+        expected = Path(f"{test_dir.realpath()}/{'SomeFile.h'}")
 
-    def test_function_and_macro_calls(self, tmpdir):
-        preprocess_and_compare('function_and_macro_calls.cpp', 'function_and_macro_calls.cpp.preprocessed.cc', tmpdir)  # NOQA
+        assert(found is not False)
```

Comment:
question on assert, assuming it just evaluates an expression, so if you just did 
`assert( found  ) `
would that not work the same?

Reply:
Ooh, good catch. Updating.